src/m365/spo/commands/homesite/homesite-set.ts

```diff
@@ -77,26 +70,151 @@ class SpoHomeSiteSetCommand extends SpoCommand {
+
+      if (args.options.vivaConnectionsDefaultStart !== undefined ||
+        args.options.draftMode !== undefined ||
+        args.options.audienceIds !== undefined ||
+        args.options.audienceNames !== undefined ||
+        args.options.targetedLicenseType !== undefined ||
+        args.options.order !== undefined) {
+
+        const configuration: any = {};
+        if (args.options.vivaConnectionsDefaultStart !== undefined) {
+          configuration.IsVivaConnectionsDefaultStartPresent = true;
+          configuration.vivaConnectionsDefaultStart = args.options.vivaConnectionsDefaultStart;
+        }
+        if (args.options.draftMode !== undefined) {
+          configuration.IsInDraftModePresent = true;
+          configuration.isInDraftMode = args.options.draftMode;
+        }
+        if (args.options.audienceIds !== undefined) {
+          configuration.IsAudiencesPresent = true;
+          configuration.Audiences = args.options.audienceIds.split(',').map(id => id.trim());
+        }
+        if (args.options.audienceNames !== undefined) {
+          configuration.IsAudiencesPresent = true;
+          configuration.Audiences = await this.transformAudienceNamesToIds(args.options.audienceNames);
+        }
+        if (args.options.targetedLicenseType !== undefined) {
+          configuration.IsTargetedLicenseTypePresent = true;
+          configuration.TargetedLicenseType = this.convertTargetedLicenseTypeToNumber(args.options.targetedLicenseType);
+        }
+        if (args.options.order !== undefined) {
+          configuration.IsOrderPresent = true;
+          configuration.Order = args.options.order;
+        }
+        const hasOnlyVivaConnectionsDefaultStart = args.options.vivaConnectionsDefaultStart !== undefined &&
+          args.options.draftMode === undefined &&
+          args.options.audienceIds === undefined &&
+          args.options.audienceNames === undefined &&
+          args.options.targetedLicenseType === undefined &&
+          args.options.order === undefined;
+        if (homeSiteCount <= 1 && hasOnlyVivaConnectionsDefaultStart) {
+          requestOptions.url += '/SetSPHSiteWithConfiguration';
+          requestOptions.data.sphSiteUrl = args.options.siteUrl;
+          requestOptions.data.configuration = { vivaConnectionsDefaultStart: args.options.vivaConnectionsDefaultStart };;
+        }
+        else {
+          requestOptions.url += '/UpdateTargetedSite';
+          requestOptions.data.siteUrl = args.options.siteUrl;
+          requestOptions.data.configurationParam = configuration;
+        }
       }
-      else {
+      else if (homeSiteCount <= 1) {
         requestOptions.url += '/SetSPHSite';
+        requestOptions.data.sphSiteUrl = args.options.siteUrl;
       }
 
-      const res = await request.post<{ value: string; }>(requestOptions);
-      await logger.log(res.value);
+      const res = await request.post(requestOptions);
+      await logger.log(res);
```

Comment:
Currently, for the old API, I'm getting the following response. Let's ensure that the response does not change as this is a breaking change.

```
{
  "value": "The Home site has been set to https://milanhdev.sharepoint.com. It may take some time for the change to apply. Check aka.ms/homesites for details."
}
```

Reply:
That's the experience using the old endpoints, this is an example using the current cli for m365 without the changes of this PR
<img width="1203" height="132" alt="image" src="https://github.com/user-attachments/assets/da70fd68-c1df-4f70-8c4d-a1697617f882" />

Reply:
Yes true, but in the current implementation, we just have the notification message. Now this message is wrapped into an object with value property.